ebpf/src/bcc_probe_generator.cpp

```diff
@@ -0,0 +1,300 @@
+}
+
+osquery::Status getOsqueryPidList(std::vector<pid_t>& pid_list) {
+  pid_list.clear();
+
+  boostfs::path proc_folder("/proc");
+
+  for (const auto& entry : boost::make_iterator_range(
+           boostfs::directory_iterator(proc_folder), {})) {
+    if (!boostfs::is_directory(entry)) {
+      continue;
+    }
+
+    auto process_com_path = entry / "comm";
+
+    std::ifstream comm_stream(process_com_path.string());
+
+    std::string process_name((std::istreambuf_iterator<char>(comm_stream)),
+                             std::istreambuf_iterator<char>());
+    process_name.pop_back();
+
+    if (process_name != "osqueryi" && process_name != "osqueryd") {
+      continue;
+    }
+
+    auto pid =
+        osquery::tryTo<pid_t>(entry.path().filename().string(), 10).takeOr(0);
+
+    if (pid != 0) {
+      pid_list.push_back(pid);
+    }
+  }
+
+  if (pid_list.empty()) {
+    return osquery::Status::failure("No running osquery found");
+  }
+
+  return osquery::Status(0);
+}
+
+osquery::Status generateManagedTracepointProbeSource(
+    std::string& probe_source_code, const ManagedTracepointProbe& desc) {
+  probe_source_code.clear();
+
+  std::vector<pid_t> osquery_pid_list;
+  auto status = getOsqueryPidList(osquery_pid_list);
+  if (!status.ok()) {
+    return status;
+  }
+
+  osquery_pid_list.push_back(getpid());
+
+  std::stringstream buffer;
+
+  buffer << "#define STRING_BUFFER_SIZE " << desc.string_buffer_size << "\n";
+  buffer << "#define STRING_LIST_SIZE " << desc.string_list_size << "\n\n";
+
+  buffer << kBccProbeAPI << "\n";
+
+  std::size_t tracepoint_id = 0U;
+  for (const auto& tracepoint_desc : desc.tracepoint_list) {
+    auto tracepoint_handler = generateTracepointHandler(tracepoint_desc,
+                                                        tracepoint_id++,
+                                                        desc.string_buffer_size,
+                                                        osquery_pid_list);
+
+    buffer << tracepoint_handler;
+  }
+
+  probe_source_code = buffer.str();
+  return osquery::Status(0);
+}
+} // namespace
+
+osquery::Status generateManagedTracepointProbe(
+    eBPFProbeRef& probe, const ManagedTracepointProbe& desc) {
+  std::string probe_source_code;
+  auto status = generateManagedTracepointProbeSource(probe_source_code, desc);
+  if (!status.ok()) {
+    return status;
+  }
+
+  eBPFProbeDescriptor probe_descriptor;
+  probe_descriptor.name = desc.name;
+  probe_descriptor.source_code = probe_source_code;
+
+  for (const auto& tracepoint : desc.tracepoint_list) {
+    eBPFProbeDescriptor::Probe probe = {};
+    probe.type = eBPFProbeDescriptor::Probe::Type::Tracepoint;
+    probe.entry = (tracepoint.name.find("sys_enter_") != std::string::npos);
+    probe.translate_name = true;
+    probe.name = tracepoint.name;
+
+    probe_descriptor.probe_list.push_back(std::move(probe));
+  }
+
+  status = eBPFProbe::create(probe, probe_descriptor);
+  if (!status.ok()) {
```

Comment:
Unused check? We fall through to returning `Status(0)`, so maybe just an unconditional `return status` here? Ignore if the plan is to add a `TLOG` or something else here 😄